### PR TITLE
refactor/use-errors

### PR DESCRIPTION
**Pull request by ghost**
refactor(ui,sdk): update error handling and logging to rely on react-query

chore(): remove useErrors and useLoginState hooks

docs(): update README

## Description
* Refactored most hook queries and mutations to depend on the onError handler for logging instead of try/catch
* catchError operator used for cases when recovery from error made sense
* ErrorInfoCard component refactored to use directly the error property from useQuery
* Few fixes where logging was non-existent or was implemented but did not work correctly
* Removed now unused use-login-state and use-error-state hooks
* Updated hooks package README

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/hooks/src/use-moderation.ts [hunk 1]
```diff
@@ -45,34 +45,28 @@ export type UseModerationParam = {
...
+  const status = await createModeration(url, data);
+
+  switch (true) {
+    case status === 409:
+      throw new Error(`This content has already been moderated by you`);
+    case status === 403:
+      throw new Error('You are not authorized to perform this operation');
+    case status === 400:
+      throw new Error('Bad request. Please try again later');
+    case status >= 400:
```
Comment (kenshyx) [inline ui/hooks/src/use-moderation.ts:63]: you can remove both checks for ( === or > 400) and replace it with default behaviour

File: ui/hooks/src/use-moderation.ts [hunk 2]
```diff
@@ -99,40 +93,37 @@ function useModeration() {
...
+  const status = await createModeration(url, data);
+
+  switch (true) {
+    case status === 409:
+      throw new Error(`This content has already been reported by you`);
+    case status === 403:
+      throw new Error('You are not authorized to perform this operation');
+    case status === 400:
+      throw new Error('Bad request. Please try again later');
+    case status >= 400:
```
Comment (kenshyx) [inline ui/hooks/src/use-moderation.ts:118]: same as my previous comment

File: ui/hooks/src/use-moderation.ts (hunk 1 above)
Comment (ghost) [reply ui/hooks/src/use-moderation.ts:63]: I removed the === 400 case, but I'm not sure 'throw Error' should be the default behaviour :D

Comment (kenshyx) [reply ui/hooks/src/use-moderation.ts:63]: hmm, yes, I didn't see that there is no success case

Comment (kenshyx) [reply ui/hooks/src/use-moderation.ts:63]: then the `return status` part should be in default case

**Review by kenshyx — APPROVED** [review]

**Pull request merged**